client/go/dogma/cmd/edit_cmd.go

```diff
@@ -0,0 +1,121 @@
+// Copyright 2018 LINE Corporation
+//
+// LINE Corporation licenses this file to you under the Apache License,
+// version 2.0 (the "License"); you may not use this file except in compliance
+// with the License. You may obtain a copy of the License at:
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations
+// under the License.
+
+package cmd
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path"
+
+	"github.com/line/centraldogma/client/go/dogma"
+	"github.com/urfave/cli"
+)
+
+// An editCommand modifies the file of the specified path with the revision.
+type editFileCommand struct {
+	repo repositoryRequestInfo
+}
+
+func (ef *editFileCommand) execute(c *cli.Context) error {
+	repo := ef.repo
+	remoteEntry, err := getRemoteFileEntry(
+		c, repo.remoteURL, repo.proName, repo.repoName, repo.path, repo.revision, "")
+	if err != nil {
+		return err
+	}
+	change, err := editRemoteFileContent(remoteEntry)
+	if err != nil {
+		return err
+	}
+
+	commitMessage, err := getCommitMessage(c, change.Path, edition)
+	if err != nil {
+		return err
+	}
+
+	client, err := newDogmaClient(c, repo.remoteURL)
+	if err != nil {
+		return err
+	}
+
+	_, res, err := client.Push(context.Background(),
+		repo.proName, repo.repoName, repo.revision, commitMessage, []*dogma.Change{change})
```

Comment:
Indentation seems weird. Is this what gofmt suggests?

Reply:
No, it was wrong :-)